pallets/schemas/src/lib.rs

```diff
@@ -398,23 +410,46 @@ pub mod pallet {
 					set_settings.set(i);
 				}
 			}
-			let schema = Schema { model_type, model, payload_location, settings: set_settings };
+			let schema_info = SchemaInfo { model_type, payload_location, settings: set_settings };
 			<CurrentSchemaIdentifierMaximum<T>>::set(schema_id);
-			<Schemas<T>>::insert(schema_id, schema);
+			<SchemaInfos<T>>::insert(schema_id, schema_info);
+			<SchemaPayloads<T>>::insert(schema_id, model);
 			Ok(schema_id)
 		}
 
 		/// Retrieve a schema by id
 		pub fn get_schema_by_id(schema_id: SchemaId) -> Option<SchemaResponse> {
```

Comment:
Hmm, I'm thinking there are use cases where you _only_ want to get the model/payload, and not the other info... perhaps add another method & RPC `get_schema_model_by_id`. Then we could decide whether we need this one, depending on our analysis of use cases. I think in a lot of use cases you only need one or the other.

Reply:
+1

Reply:
I think there is confusion going on about RPCs and methods for this. Right now we have an RPC that returns the `SchemaResponse` which is the combination of payload and metadata. I don't think there is any need for an RPC that only returns payload since we don't need to optimize queries that we expose to RPC since all of it is considered off-chain processing. Adding an exposed method that only returns payload would only make sense if there was any need to query that on-chain which currently we don't.

Reply:
True, maybe that's the subject of a separate issue/PR that is focused on RPC network efficiency for client interactions.